src/chain.rs

```diff
@@ -333,3 +413,55 @@ pub mod kusama {
 		};
 	}
 }
+
+/// Helper to fetch the weight from a remote node
+///
+/// Panics: if the RPC call fails or if decoding the response as a `Weight` fails.
+fn get_weight<T: Encode>(tx: subxt::tx::StaticTxPayload<T>) -> Weight {
+	futures::executor::block_on(async {
+		let client = SHARED_CLIENT.get().expect("shared client is configured as start; qed");
+
+		let call_data = {
+			let mut buffer = Vec::new();
+
+			let encoded_call = client.tx().call_data(&tx).unwrap();
+			let encoded_len = encoded_call.len() as u32;
+
+			buffer.extend(encoded_call);
+			encoded_len.encode_to(&mut buffer);
+
+			Bytes(buffer)
+		};
+
+		let bytes: Bytes = client
+			.rpc()
+			.client
+			.request(
+				"state_call",
+				rpc_params!["TransactionPaymentCallApi_query_call_info", call_data],
+			)
+			.await
+			.unwrap();
+
+		let info: RuntimeDispatchInfo<Balance> = Decode::decode(&mut bytes.0.as_ref()).unwrap();
+
+		log::debug!(
+			target: LOG_TARGET,
+			"Received weight of `Solution Extrinsic` from remote node: {:?}",
+			info.weight
+		);
+
+		info.weight
+	})
+}
+
+fn mock_votes(voters: u32, desired_targets: u16) -> Vec<(u32, u16)> {
+	assert!(voters >= desired_targets as u32);
```

Comment:
@kianenigma 

I guess this must be true otherwise won't "all desired targets" be used